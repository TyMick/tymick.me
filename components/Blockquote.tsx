import React from "react";
import { css, useTheme } from "@emotion/react";
import clsx from "clsx";

export default function Blockquote({
  className,
  ...props
}: React.ComponentPropsWithoutRef<"blockquote">) {
  const theme = useTheme();

  return (
    <blockquote
      className={clsx(className, "pl-2 my-4")}
      css={css`
        border-left: 4px solid ${theme.colors.gray500};
      `}
      {...props}
    />
  );
}
