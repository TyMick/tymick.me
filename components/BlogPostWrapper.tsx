import React from "react";
import Link from "next/link";
import { useRouter } from "next/router";
import { NextSeo } from "next-seo";
import { Container } from "react-bootstrap";
import useWindowWidthBreakpoints from "use-window-width-breakpoints";
import { get, last as lastElement, startCase } from "lodash";
import clsx from "clsx";
import DateTime from "./DateTime";
import { parseInlineMarkdown } from "../lib/text-processing";

type BlogPostWrapperProps = {
  title: string;
  subtitle?: string;
  description?: string;
  /** Date/time first published, in ISO format. */
  date: string;
  /** Date/time last updated, in ISO format. */
  lastUpdated?: string;
  canonicalUrl?: string;
  ogImage?: {
    filename: string;
    alt?: string;
  };
  cta?: string;
  socialLinks?: {
    twitter?: string;
    mastodon?: string;
    facebook?: string;
    linkedin?: string;
  };
  ipynb?: boolean;
  className?: string;
  [propName: string]: any;
};

export default function BlogPostWrapper({
  title,
  subtitle,
  description,
  date,
  lastUpdated,
  canonicalUrl,
  ogImage,
  cta,
  socialLinks,
  ipynb,
  className,
  ...otherProps
}: BlogPostWrapperProps) {
  const breakpoint = useWindowWidthBreakpoints();

  const numLinks: number = socialLinks
    ? Object.keys(socialLinks).length
    : undefined;
  function linkSort(
    [keyA, valueA]: [string, string],
    [keyB, valueB]: [string, string]
  ): number {
    const linkRanks = { twitter: 1, mastodon: 2, facebook: 3, linkedin: 4 };
    return get(linkRanks, keyA, 100) - get(linkRanks, keyB, 100);
  }

  const router = useRouter();
  const slug: string = lastElement(router.pathname.split("/"));
  const url = `https://tymick.me/blog/${slug}`;
  const webPageTitle = `${title} - Ty Mick`;

  return (
    <>
      <NextSeo
        {...{
          title: webPageTitle,
          description: description || subtitle,
          canonical: canonicalUrl || url,
          openGraph: {
            title: webPageTitle,
            description: description || subtitle,
            url: url,
            images: ogImage
              ? [
                  {
                    url: `https://tymick.me/images/${ogImage.filename}`,
                    alt: ogImage?.alt,
                  },
                ]
              : undefined,
            site_name: "Ty Mick",
          },
          twitter: {
            cardType: "summary_large_image",
            handle: "@tywmick",
            site: "@tywmick",
          },
        }}
      />

      <Container
        as="article"
        aria-labelledby="article-title"
        className="serif cap-width-lg"
      >
        <h1 id="article-title">{parseInlineMarkdown(title)}</h1>

        {subtitle && (
          <div className="h3 text-secondary">
            {parseInlineMarkdown(subtitle)}
          </div>
        )}

        <div className="font-italic text-secondary sans-serif mb-4">
          <DateTime date={date} formatString="LLLL d, yyyy" />
          {lastUpdated && (
            <>
              {breakpoint.xs ? <br /> : " "}
              (last updated{" "}
              <DateTime date={lastUpdated} formatString="LLLL d, yyyy" />)
            </>
          )}
        </div>

        <div
          className={clsx(className, "markdown-post", ipynb && "ipynb")}
          {...otherProps}
        />

        {socialLinks && (
          <>
            <hr />
            <p>
              {cta
                ? parseInlineMarkdown(cta)
                : "Want to discuss this topic further? Chime in"}{" "}
              on{" "}
              {Object.entries(socialLinks)
                .sort(linkSort)
                .map(([label, link], i) => (
                  <span key={label}>
                    <a href={link} className={`${label}-link`}>
                      {label === "linkedin" ? "LinkedIn" : startCase(label)}
                    </a>

                    {
                      // Determine appropriate list punctuation
                      numLinks === 1
                        ? ""
                        : numLinks === 2
                        ? i + 1 === numLinks
                          ? ""
                          : " or "
                        : i + 1 === numLinks
                        ? ""
                        : i + 2 === numLinks
                        ? ", or "
                        : ", "
                    }
                  </span>
                ))}
              .
            </p>
          </>
        )}

        <hr className="border-secondary" />

        <p className="text-secondary font-italic">
          Found an error or typo in this post you&rsquo;d like to fix? Send me a{" "}
          <a
            href={`https://github.com/TyMick/tymick.me/edit/master/pages/blog/${slug}.mdx`}
          >
            pull request on GitHub
          </a>
          !
        </p>

        <p className="text-secondary font-italic">
          Want to publish this article on your blog, in your magazine, or
          anywhere else? This post, like all the content on my website, is
          licensed under a{" "}
          <a href="https://creativecommons.org/licenses/by/4.0/">
            Creative Commons Attribution license
          </a>
          , so you&rsquo;re welcome to share it wherever and however you please,
          as long as you cite me as the author. I&rsquo;d also enjoy{" "}
          <Link href="/connect">
            <a>hearing from you</a>
          </Link>{" "}
          if you do publish this somewhere, but that&rsquo;s totally up to you.
        </p>
      </Container>
    </>
  );
}

export function withBlogPostWrapper(wrapperProps: BlogPostWrapperProps) {
  return ({ children }) => (
    <BlogPostWrapper {...wrapperProps}>{children}</BlogPostWrapper>
  );
}
