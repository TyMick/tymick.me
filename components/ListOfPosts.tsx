import { trimEnd } from "lodash";
import React from "react";
import { parseInlineMarkdown } from "../lib/text-processing";
import EssayPreview from "./EssayPreview";

export type Post = {
  title: string;
  subtitle?: string;
  slug: string;
  /** Date/time first published, in ISO format. */
  date: string;
  excerpt: string;
};

export type Props = {
  posts: Post[];
};

export default function ListOfPosts({ posts }: Props) {
  return (
    <>
      <ul className="list-unstyled mb-n4">
        {posts.map(({ title, subtitle, slug, date, excerpt }) => (
          <EssayPreview
            title={title}
            subtitle={subtitle}
            href={`/blog/${slug}`}
            date={date}
            key={slug}
          >
            {parseInlineMarkdown(trimEnd(excerpt, ".") + "...")}
          </EssayPreview>
        ))}
      </ul>
    </>
  );
}
