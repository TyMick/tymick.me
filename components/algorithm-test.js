/**
 * Copyright 2020 Tyler Westin Mick
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React, { useState } from "react";
import { Form, InputGroup, Card } from "react-bootstrap";
import SyntaxHighlighter from "react-syntax-highlighter";
import { githubGist } from "react-syntax-highlighter/dist/cjs/styles/hljs";

const AlgorithmTest = ({ algorithm, argumentName, inputType }) => {
  const [argValue, setArgValue] = useState(null);

  return (
    <div className="mx-2 mx-sm-5 mb-3">
      <Form inline className="mb-3 text-monospace">
        <InputGroup className="w-100">
          <InputGroup.Prepend>
            <InputGroup.Text id="functionName">
              {algorithm.name}({inputType === "text" && argValue && '"'}
            </InputGroup.Text>
          </InputGroup.Prepend>
          <Form.Control
            type={inputType}
            placeholder={argumentName}
            aria-label={argumentName}
            aria-describedby="functionName"
            onChange={(e) => {
              setArgValue(e.target.value);
            }}
          />
          <InputGroup.Append>
            <InputGroup.Text>
              {inputType === "text" && argValue && '"'})
            </InputGroup.Text>
          </InputGroup.Append>
        </InputGroup>
      </Form>

      <div className="d-flex align-items-center">
        <h5 className="mb-0 mr-3">Returns:</h5>
        <Card className="h5 mb-0">
          {argValue ? (
            <SyntaxHighlighter
              language="javascript"
              style={githubGist}
              className="px-2 py-1 m-0"
            >
              {typeof algorithm(argValue) === "string"
                ? '"' + algorithm(argValue) + '"'
                : algorithm(argValue).toString()}
            </SyntaxHighlighter>
          ) : (
            <pre className="px-2 py-1 m-0">&nbsp;</pre>
          )}
        </Card>
      </div>
    </div>
  );
};

export default AlgorithmTest;
