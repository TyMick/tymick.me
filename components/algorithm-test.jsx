"use strict";

import React, { useState } from "react";
import { Form, InputGroup, Card } from "react-bootstrap";
import SyntaxHighlighter from "react-syntax-highlighter";
import { githubGist } from "react-syntax-highlighter/dist/cjs/styles/hljs";

export default ({ algorithm, argumentName, inputType }) => {
  const [argValue, setArgValue] = useState(null);

  return (
    <div className="mx-2 mx-sm-5 mb-3">
      <Form inline className="mb-3">
        <InputGroup>
          <InputGroup.Prepend>
            <InputGroup.Text id="functionName">
              {algorithm.name}({inputType === "text" && argValue && '"'}
            </InputGroup.Text>
          </InputGroup.Prepend>
          <Form.Control
            type={inputType}
            placeholder={argumentName}
            aria-label={argumentName}
            aria-describedby="functionName"
            onChange={e => {
              setArgValue(e.target.value);
            }}
          />
          <InputGroup.Append>
            <InputGroup.Text>
              {inputType === "text" && argValue && '"'})
            </InputGroup.Text>
          </InputGroup.Append>
        </InputGroup>
      </Form>

      <div className="d-flex align-items-center">
        <h5 className="mb-0 mr-3">Returns:</h5>
        <Card className="h5 mb-0">
          {argValue ? (
            <SyntaxHighlighter
              language="javascript"
              style={githubGist}
              className="px-2 py-1 m-0"
            >
              {typeof algorithm(argValue) === "string"
                ? '"' + algorithm(argValue) + '"'
                : algorithm(argValue).toString()}
            </SyntaxHighlighter>
          ) : (
            <pre className="px-2 py-1 m-0">&nbsp;</pre>
          )}
        </Card>
      </div>
    </div>
  );
};
