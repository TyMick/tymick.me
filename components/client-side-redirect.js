import React, { useEffect } from "react";
import { useRouter } from "next/router";
import Head from "next/head";
import { Container } from "react-bootstrap";

export default function Redirect({ url, as }) {
  const router = useRouter();

  useEffect(() => {
    if (as) {
      router.push(url, as);
    } else if (url[0] === "/") {
      router.push(url);
    } else {
      window.location = url;
    }
  }, []);

  return (
    <>
      <Head>
        <noscript>
          <meta httpEquiv="refresh" content={`0;url=${url}`} />
        </noscript>
        <link rel="canonical" href={url} />
      </Head>
      <Container fluid as="p" style={{ textAlign: "left" }}>
        Redirecting to <a href={url}>{url}</a>&hellip;
      </Container>
    </>
  );
}
