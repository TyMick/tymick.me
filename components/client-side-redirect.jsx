import React, { useEffect } from "react";
import Head from "next/head";
import { Container } from "react-bootstrap";

export default function Redirect({ href }) {
  useEffect(() => {
    window.location = href;
  }, []);

  return (
    <>
      <Head>
        <noscript>
          <meta httpEquiv="refresh" content={`0;url=${href}`} />
        </noscript>
        <link rel="canonical" href={href} />
      </Head>
      <Container fluid as="p" style={{ textAlign: "left" }}>
        Redirecting to <a href={href}>{href}</a>&hellip;
      </Container>
    </>
  );
}
