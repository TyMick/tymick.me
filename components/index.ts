import { Components } from "@mdx-js/react";
import { withMarkdownParsing } from "../lib/text-processing";
import AlgorithmTest from "./AlgorithmTest";
import Blockquote from "./Blockquote";
import BlogPostWrapper, { withBlogPostWrapper } from "./BlogPostWrapper";
import Codepen from "./Codepen";
import ContactMethod from "./ContactMethod";
import DateTime from "./DateTime";
import EssayPreview from "./EssayPreview";
import { Figure, FigureImage, FigureCaption } from "./Figure";
import Footer from "./Footer";
import Image from "./Image";
import InlineFootnote from "./InlineFootnote";
import Link from "./Link";
import MiscProfile from "./MiscProfile";
import Nav from "./Nav";
import PortfolioSection from "./PortfolioSection";
import PullQuote from "./PullQuote";
import Testimonial from "./Testimonial";
import ThematicBreak from "./ThematicBreak";

export {
  AlgorithmTest,
  Blockquote,
  BlogPostWrapper,
  withBlogPostWrapper,
  Codepen,
  ContactMethod,
  DateTime,
  EssayPreview,
  Figure,
  FigureImage,
  FigureCaption,
  Footer,
  Image,
  InlineFootnote,
  Link,
  MiscProfile,
  Nav,
  PortfolioSection,
  PullQuote,
  Testimonial,
  ThematicBreak,
};

export const mdxComponents: Components & {
  /**
   * @todo Find a better typing for this. I tried React.ComponentType<any>, but
   * that doesn't work with `Figure`. I can't find anything better at the
   * moment.
   */
  [shortcode: string]: any;
} = {
  // Component replacements for Markdown syntax
  a: Link,
  blockquote: Blockquote,
  hr: ThematicBreak,

  // Shortcodes (components available in every MDX file)
  Blockquote: withMarkdownParsing(Blockquote),
  Figure,
  FigureImage,
  FigureCaption: withMarkdownParsing(FigureCaption),
  I: withMarkdownParsing("i"),
  Image,
  InlineFootnote: withMarkdownParsing(InlineFootnote),
  LI: withMarkdownParsing("li"),
  P: withMarkdownParsing("p"),
  PullQuote: withMarkdownParsing(PullQuote),

  // Layout component
  /**
   * Currently not working when using MDX files as Next.js pages, as the {@link https://mdxjs.com/getting-started#using-the-wrapper
   * wrapper component} doesn't receive {@link https://mdxjs.com/advanced/components#layout-props
   * `layoutProps`} as it should. Perhaps this will be fixed in future versions
   * of MDX and/or @next/mdx?
   *
   * @todo Open an issue in MDX and/or Next.js.
   */
  // wrapper: BlogPostWrapper,
};
