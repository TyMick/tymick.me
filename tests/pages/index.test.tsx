import React from "react";
import renderer from "react-test-renderer";
import { PostsProps } from "../../components/ListOfPosts";
import Home, { getStaticProps } from "../../pages/index";
import { SuccessfulPropsResult } from "../util";

describe("Home", () => {
  it("matches snapshot", async () => {
    const { props } = (await getStaticProps(
      {}
    )) as SuccessfulPropsResult<PostsProps>;
    expect(props).toBeDefined();
    expect(renderer.create(<Home {...props} />).toJSON()).toMatchSnapshot();
  });
});
