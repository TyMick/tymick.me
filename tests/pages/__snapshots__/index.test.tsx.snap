// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
[
  <div
    className="cap-lg-49 container"
  >
    <div
      className="align-items-center mb-3 mb-sm-5 row"
    >
      <div
        className="mr-md-4 mb-2 mb-sm-0 text-center text-sm-left col-sm-auto col-12"
      >
        <div
          className="Image__ImageWrapper-sc-wuromz-0 divnQl image-wrapper"
          width="100%"
        >
          <span
            style={
              {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                style={
                  {
                    "background": "none",
                    "border": 0,
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              />
            </span>
            <img
              alt="Ty hiking in Joshua Tree National Park"
              className="rounded-circle"
              data-nimg="intrinsic"
              decoding="async"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
              style={
                {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </span>
        </div>
      </div>
      <div
        className="text-center text-sm-left col-sm col-12"
      >
        <h1
          className="greeting"
        >
          Hi! I’m Ty.
        </h1>
      </div>
    </div>
    <ul
      className="list-unstyled mb-n4"
    >
      <li
        className="mb-5"
      >
        <div
          className="DateHeading___StyledDiv-sc-17zqdwo-0 cilEMx"
        >
          October 31, 2022
        </div>
        <div
          className="position-relative"
        >
          <h2
            className="h5 serif"
          >
            <a
              className="text-reset stretched-link"
              href="/blog/whatsmyname-puts-drivers-at-risk"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              whatsmyname-puts-drivers-at-risk title
            </a>
          </h2>
          <h3
            className="h6 serif text-muted font-italic"
          >
            whatsmyname-puts-drivers-at-risk subtitle
          </h3>
          <p
            className="serif"
          >
            whatsmyname-puts-drivers-at-risk excerpt…
          </p>
        </div>
      </li>
      <li
        className="mb-5"
      >
        <div
          className="DateHeading___StyledDiv-sc-17zqdwo-0 cilEMx"
        >
          October 31, 2022
        </div>
        <div
          className="position-relative"
        >
          <h2
            className="h5 serif"
          >
            <a
              className="text-reset stretched-link"
              href="/blog/sunshine-protection-act"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              sunshine-protection-act title
            </a>
          </h2>
          <h3
            className="h6 serif text-muted font-italic"
          >
            sunshine-protection-act subtitle
          </h3>
          <p
            className="serif"
          >
            sunshine-protection-act excerpt…
          </p>
        </div>
      </li>
      <li
        className="mb-5"
      >
        <div
          className="DateHeading___StyledDiv-sc-17zqdwo-0 cilEMx"
        >
          October 31, 2022
        </div>
        <div
          className="position-relative"
        >
          <h2
            className="h5 serif"
          >
            <a
              className="text-reset stretched-link"
              href="/blog/optimal-investing"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              optimal-investing title
            </a>
          </h2>
          <h3
            className="h6 serif text-muted font-italic"
          >
            optimal-investing subtitle
          </h3>
          <p
            className="serif"
          >
            optimal-investing excerpt…
          </p>
        </div>
      </li>
      <li
        className="mb-5"
      >
        <div
          className="DateHeading___StyledDiv-sc-17zqdwo-0 cilEMx"
        >
          October 31, 2022
        </div>
        <div
          className="position-relative"
        >
          <h2
            className="h5 serif"
          >
            <a
              className="text-reset stretched-link"
              href="/blog/new-job-new-site-design"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              new-job-new-site-design title
            </a>
          </h2>
          <h3
            className="h6 serif text-muted font-italic"
          >
            new-job-new-site-design subtitle
          </h3>
          <p
            className="serif"
          >
            new-job-new-site-design excerpt…
          </p>
        </div>
      </li>
      <li
        className="mb-5"
      >
        <div
          className="DateHeading___StyledDiv-sc-17zqdwo-0 cilEMx"
        >
          October 31, 2022
        </div>
        <div
          className="position-relative"
        >
          <h2
            className="h5 serif"
          >
            <a
              className="text-reset stretched-link"
              href="/blog/loan-risk-nlp"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              loan-risk-nlp title
            </a>
          </h2>
          <h3
            className="h6 serif text-muted font-italic"
          >
            loan-risk-nlp subtitle
          </h3>
          <p
            className="serif"
          >
            loan-risk-nlp excerpt…
          </p>
        </div>
      </li>
      <li
        className="mb-5"
      >
        <div
          className="DateHeading___StyledDiv-sc-17zqdwo-0 cilEMx"
        >
          October 31, 2022
        </div>
        <div
          className="position-relative"
        >
          <h2
            className="h5 serif"
          >
            <a
              className="text-reset stretched-link"
              href="/blog/loan-risk-neural-network"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              loan-risk-neural-network title
            </a>
          </h2>
          <h3
            className="h6 serif text-muted font-italic"
          >
            loan-risk-neural-network subtitle
          </h3>
          <p
            className="serif"
          >
            loan-risk-neural-network excerpt…
          </p>
        </div>
      </li>
      <li
        className="mb-5"
      >
        <div
          className="DateHeading___StyledDiv-sc-17zqdwo-0 cilEMx"
        >
          October 31, 2022
        </div>
        <div
          className="position-relative"
        >
          <h2
            className="h5 serif"
          >
            <a
              className="text-reset stretched-link"
              href="/blog/loan-grading-showdown"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              loan-grading-showdown title
            </a>
          </h2>
          <h3
            className="h6 serif text-muted font-italic"
          >
            loan-grading-showdown subtitle
          </h3>
          <p
            className="serif"
          >
            loan-grading-showdown excerpt…
          </p>
        </div>
      </li>
      <li
        className="mb-5"
      >
        <div
          className="DateHeading___StyledDiv-sc-17zqdwo-0 cilEMx"
        >
          October 31, 2022
        </div>
        <div
          className="position-relative"
        >
          <h2
            className="h5 serif"
          >
            <a
              className="text-reset stretched-link"
              href="/blog/debug-nextjs-with-vs-code"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              debug-nextjs-with-vs-code title
            </a>
          </h2>
          <h3
            className="h6 serif text-muted font-italic"
          >
            debug-nextjs-with-vs-code subtitle
          </h3>
          <p
            className="serif"
          >
            debug-nextjs-with-vs-code excerpt…
          </p>
        </div>
      </li>
      <li
        className="mb-5"
      >
        <div
          className="DateHeading___StyledDiv-sc-17zqdwo-0 cilEMx"
        >
          October 31, 2022
        </div>
        <div
          className="position-relative"
        >
          <h2
            className="h5 serif"
          >
            <a
              className="text-reset stretched-link"
              href="/blog/covid-19-social-distancing"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              covid-19-social-distancing title
            </a>
          </h2>
          <h3
            className="h6 serif text-muted font-italic"
          >
            covid-19-social-distancing subtitle
          </h3>
          <p
            className="serif"
          >
            covid-19-social-distancing excerpt…
          </p>
        </div>
      </li>
      <li
        className="mb-5"
      >
        <div
          className="DateHeading___StyledDiv-sc-17zqdwo-0 cilEMx"
        >
          October 31, 2022
        </div>
        <div
          className="position-relative"
        >
          <h2
            className="h5 serif"
          >
            <a
              className="text-reset stretched-link"
              href="/blog/console-log-after-reload"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              console-log-after-reload title
            </a>
          </h2>
          <h3
            className="h6 serif text-muted font-italic"
          >
            console-log-after-reload subtitle
          </h3>
          <p
            className="serif"
          >
            console-log-after-reload excerpt…
          </p>
        </div>
      </li>
    </ul>
  </div>,
  <a
    href="https://mastodon.social/@TyMick"
    rel="me"
  />,
]
`;
